### DISCUSSION TITLE
Package mockoon cli with android app

### DISCUSSION THREAD

**Discussion by susonthapa** [mockoon/mockoon#558] [Ideas] [answered]
It might seem crazy but I am trying to package mockoon cli with android app. The usecase is to embed mockoon in android app so that our clients don't have to run mockoon separately or I don't have to host it somewhere. I just want the client to view the mocked app without internet connection so that he can get the feel of it.

What I have accomplished so far?
I am able to run node js code in android and I can launch mockoon from there. But when I try to start a environment it tries to start another process which is not allowed in android (as far as I know). This is the screenshot of the problem.

![image](https://user-images.githubusercontent.com/33973551/131773406-4c3711ac-7a9b-4c66-bd68-3bdb6640bfda.png)

I am just wondering if it is possible to run mockoon on the same process ( I run node on separate thread) and prevent mockoon from trying to write into read only filesystem. 
What I am trying to do is a very niche usecase and I know there are many options available. I just wanted to know if this can be done. Thank you.

**Comment by 255kb** [comment, marked as answer]
Hi @susonthapa First, thanks for your medium article. It's always great to see what the community uses Mockoon for. 😄 

Concerning your use case, there is currently no way to use the CLI without delegating the processes management to PM2. 
What you could do instead is to use directly our lib https://github.com/mockoon/commons-server which is publish on GitHub. It accepts a Mockoon environment and directly launch the ExpressJS server. The relevant code showing how to use it is there: https://github.com/mockoon/cli/blob/main/src/libs/server.ts#L83-L91

Let me know if you have more questions.

**Reply by susonthapa** [reply]
Thank you for the information. I was able to run mockoon on android and it's working as expected. I do have a problem. The above method only works for a single environment. I tried to start multiple environments by running  multiple node js server on different threads but within a single process but it just crashed. I guess I understand why you are using PM2 as starting multiple v8 engines within a process is not allowed as far as I know. Do you have any solution for this ?

**Reply by 255kb** [reply]
I am not sure that you can run multiple Node.js HTTP servers in the same thread and I don't see another solution than running multiple threads using `spawn` or a library like https://github.com/josdejong/workerpool.

**Reply by susonthapa** [reply]
Thank you for your help. I will do my research.
